File: src/main/java/ru/serdtsev/homemoney/dto/ReserveDto.java
package ru.serdtsev.homemoney.dto;


import ru.serdtsev.homemoney.account.BalanceDto;

import java.math.BigDecimal;

public class ReserveDto extends BalanceDto {
  private BigDecimal target;

  public ReserveDto() {
  }

  public BigDecimal getTarget() {
    return target != null ? target : BigDecimal.ZERO;
  }

  public void setTarget(BigDecimal target) {
    this.target = target;
  }
}

File: src/main/kotlin/ru/serdtsev/homemoney/port/dto/account/ReserveDto.kt
package ru.serdtsev.homemoney.port.dto.account

import ru.serdtsev.homemoney.domain.model.account.AccountType
import java.math.BigDecimal
import java.time.LocalDate
import java.util.*

data class ReserveDto(
    val id: UUID,
    val name: String,
    val createdDate: LocalDate?,
    val currencyCode: String?,
    val value: BigDecimal?,
    val target: BigDecimal?,
    val isArc: Boolean?,
)